Test/lambdas/LitInt.dfy

```diff
@@ -0,0 +1,36 @@
+// RUN: %dafny /dprint:"%t.dprint" "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+const N: nat
+const M := 12;
+
+function Fibonacci(n: int): int
+{
+  if n < 2 then n
+  else Fibonacci(n - 2) + Fibonacci(n - 1)
+}
+
+function Wrong(): bool {
+  // The following assertion should fail.
+  // Specifically, the verifier should not attempt to
+  // prove that for all N, Fibonacci is not equal to 143
+  // since that would not terminate. In order to avoid that
+  // we do not wrap `const`s in `Lit`s. If we did wrap this
+  // `N` in a `Lit` the verifier would keep trying to prove this
+  // assertion.
+  assert Fibonacci(N) != 143; // should fail
+
+  // Should pass
+  assert Fibonacci(M) == 144;
+
+  // Should fail: constants are not Lit-wrapped
+  // and therefore the sum "M + 3" is not a Lit,
+  // so Fibonacci(M+5) cannot be evaluated without a fuel
+  // (but as shown belowi in `CheckFib` the verifier knows that
+  // Fibonacci(15) == 610)
+  assert Fibonacci(M + 5) == 610;
```

Comment:
`M + 5` equals 17, so this assertion does not hold according to math, so you're not testing what you claim to test? Also, typo in "belowi"

Reply:
Thanks for noticing!